lib/scheduler.js

```diff
@@ -0,0 +1,55 @@
+const fs = require('fs');
+const path = require('path');
+const tempDir = require('temp-dir');
+
+const buildPath = projectDir => path.join(tempDir, 'ava', path.basename(projectDir));
+const FILE_NAME_FAILING_TEST = 'failing-test.json';
+
+module.exports.prepareTestResultsFile = projectDir => {
+	const temporaryResultsDir = buildPath(projectDir);
+	try {
+		fs.statSync(temporaryResultsDir);
+	} catch {
+		fs.mkdirSync(temporaryResultsDir, {recursive: true});
+	}
+
+	return path.join(temporaryResultsDir, FILE_NAME_FAILING_TEST);
+};
+
+module.exports.deleteFile = filePath => {
+	try {
+		fs.unlinkSync(filePath);
+	} catch {
+	}
+};
+
+module.exports.store = (filePath, data) => {
+	fs.writeFileSync(filePath, data);
+};
+
+// Order test-files, so that files with failing tests come first
+module.exports.failingTestsFirst = (projectDir, selectedFiles) => {
+	const filePath = path.join(buildPath(projectDir), FILE_NAME_FAILING_TEST);
+	let failedTestFiles = [];
```

Comment:
No need to initialize since you return in the `catch`.

```suggestion
	let failedTestFiles;
```